Xamarin.Forms.Core/IndicatorView.cs

```diff
@@ -0,0 +1,256 @@
+
+		public Layout<View> IndicatorLayout
+		{
+			get => (Layout<View>)GetValue(IndicatorLayoutProperty);
+			set => SetValue(IndicatorLayoutProperty, value);
+		}
+
+		public int Position
+		{
+			get => (int)GetValue(PositionProperty);
+			set => SetValue(PositionProperty, value);
+		}
+
+		public int Count
+		{
+			get => (int)GetValue(CountProperty);
+			set => SetValue(CountProperty, value);
+		}
+
+		public int MaximumVisibleCount
+		{
+			get => (int)GetValue(MaximumVisibleCountProperty);
+			set => SetValue(MaximumVisibleCountProperty, value);
+		}
+
+		public DataTemplate IndicatorTemplate
+		{
+			get => (DataTemplate)GetValue(IndicatorTemplateProperty);
+			set => SetValue(IndicatorTemplateProperty, value);
+		}
+
+		public bool HidesForSingleIndicator
+		{
+			get => (bool)GetValue(HidesForSingleIndicatorProperty);
+			set => SetValue(HidesForSingleIndicatorProperty, value);
+		}
+
+		public Color IndicatorColor
+		{
+			get => (Color)GetValue(IndicatorColorProperty);
+			set => SetValue(IndicatorColorProperty, value);
+		}
+
+		public Color SelectedIndicatorColor
+		{
+			get => (Color)GetValue(SelectedIndicatorColorProperty);
+			set => SetValue(SelectedIndicatorColorProperty, value);
+		}
+
+		public double IndicatorSize
+		{
+			get => (double)GetValue(IndicatorSizeProperty);
+			set => SetValue(IndicatorSizeProperty, value);
+		}
+
+		public IEnumerable ItemsSource
+		{
+			get => (IEnumerable)GetValue(ItemsSourceProperty);
+			set => SetValue(ItemsSourceProperty, value);
+		}
+
+		void ResetIndicatorStyles()
+		{
+			try
+			{
+				BatchBegin();
+				ResetIndicatorStylesNonBatch();
+			}
+			finally
+			{
+				BatchCommit();
+			}
+		}
+
+		void ResetIndicatorCount(int oldCount)
+		{
+			try
+			{
+				BatchBegin();
+				if (oldCount < 0)
+				{
+					oldCount = 0;
+				}
+
+				if (oldCount > Count)
+				{
+					RemoveRedundantIndicatorItems();
+					return;
+				}
+
+				AddExtraIndicatorItems();
+			}
+			finally
+			{
+				ResetIndicatorStylesNonBatch();
+				BatchCommit();
+			}
+		}
+
+		void ResetIndicatorStylesNonBatch()
+		{
+			foreach (var child in Items)
+			{
+				ApplyColor(child as View);
+			}
+
+			IndicatorLayout.IsVisible = Count > 1 || !HidesForSingleIndicator;
+		}
+
+		void ResetItemsSource(IEnumerable oldItemsSource)
+		{
+			if (oldItemsSource is INotifyCollectionChanged oldObservableCollection)
+			{
+				oldObservableCollection.CollectionChanged -= OnCollectionChanged;
+			}
+
+			if (ItemsSource is INotifyCollectionChanged observableCollection)
+			{
+				observableCollection.CollectionChanged += OnCollectionChanged;
+			}
+
+			OnCollectionChanged(ItemsSource, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
+		}
+
+		void AddExtraIndicatorItems()
+		{
+			var oldCount = Items.Count;
+			for (var i = 0; i < Count - oldCount && i < MaximumVisibleCount - oldCount; ++i)
+			{
+				var size = IndicatorSize > 0 ? IndicatorSize : 10;
```

Comment:
The default value for `IndicatorSize` is -1. Why don't you make it 10? Also, you can validate/coerce it for <= 0 values so that you don't have to do it here.

Reply:
@adrianknight89 we cannot do it

@StephaneDelcroix proposed to make it -1 and it makes sense because iOS uses own "default" value, which can be different with 10.